### PR TITLE
LPD-19635 System Object definition import fails with missing Picklist field

### PR DESCRIPTION
**Pull request by diogomelo09**
[LPD-19635](https://liferay.atlassian.net/browse/LPD-19635)

### CODE REVIEW

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 1]
```diff
@@ -78,23 +78,12 @@ public static long addListTypeDefinition(
 					userId, objectField.getSystem());
 		}
 
-		ObjectFieldSetting[] objectFieldSettings = ArrayUtil.filter(
+		ObjectFieldSetting[] stateFlowSettings = ArrayUtil.filter(
```
Comment (thallescezar) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:81]: I think it is better to rename this variable to "stateFlowObjectFieldSettings". What do you think?

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 2]
```diff
@@ -103,23 +92,53 @@ public static long addListTypeDefinition(
...
+					null, userId, listTypeDefinition.getListTypeDefinitionId(),
+					key,
+					Collections.singletonMap(LocaleUtil.getDefault(), key));
 			}
+		}
 
-			listTypeEntryLocalService.addListTypeEntry(
-				null, userId, listTypeDefinition.getListTypeDefinitionId(), key,
-				Collections.singletonMap(LocaleUtil.getDefault(), key));
+		ObjectFieldSetting[] defaultValueSettings = ArrayUtil.filter(
```
Comment (thallescezar) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:122]: I think it is better to rename this variable to "defaultValueObjectFieldSettings". What do you think?

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 3]
```diff
@@ -103,23 +92,53 @@ public static long addListTypeDefinition(
...
+				ObjectFieldSettingConstants.NAME_DEFAULT_VALUE));
+
+		if (ArrayUtil.isEmpty(defaultValueSettings)) {
+			return listTypeDefinition.getListTypeDefinitionId();
 		}
 
+		String defaultValue = String.valueOf(
+			defaultValueSettings[0].getValue());
+
+		listTypeEntries.remove(defaultValue);
```
Comment (thallescezar) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:135]: Keep the pattern of previous remove method.

`if (listTypeEntries.containsKey(key)) {
					listTypeEntries.remove(key);
					continue;
				}`

**Review by thallescezar — COMMENTED** [review]
Left a few comments :)

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java (hunk 1 above)
Comment (diogomelo09) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java]: Done!! :white_check_mark:

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java (hunk 2 above)
Comment (diogomelo09) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java]: Done!! :white_check_mark:

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java (hunk 3 above)
Comment (diogomelo09) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java]: Done!! :white_check_mark:

**Comment by diogomelo09** [comment]
ci:test:sf

**Comment by diogomelo09** [comment]
ci:test:sf

**Comment by diogomelo09** [comment]
ci:test:object

**Review by thallescezar — APPROVED** [review]
LGTM!

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 4]
```diff
@@ -103,23 +92,55 @@ public static long addListTypeDefinition(
...
+			objectField.getObjectFieldSettings(),
+			objectFieldSetting -> StringUtil.equals(
+				objectFieldSetting.getName(),
+				ObjectFieldSettingConstants.NAME_DEFAULT_VALUE));
+
+		if (ArrayUtil.isEmpty(defaultValueObjectFieldSettings)) {
+			return listTypeDefinition.getListTypeDefinitionId();
 		}
 
+		String defaultValue = String.valueOf(
```
Comment (linolaoj) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:132]: Hi @diogomelo09 , can we use the `GetterUtil.getString` here?

Comment (linolaoj) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:132]: Also, would it be better to name `defaultValue` as `defaultValueObjectFieldSetting`?

**Review by linolaoj — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java (hunk 4 above)
Comment (diogomelo09) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java]: Done!! :white_check_mark:

**Comment by linolaoj** [comment]
ci:test:sf

**Comment by linolaoj** [comment]
ci:test:object

**Comment by diogomelo09** [comment]
ci:test:object

**Comment by linolaoj** [comment]
ci:test:object

**Comment by linolaoj** [comment]
Hi @diogomelo09, can you please check the tests `ObjectAutoGeneratedLayout#CanReplaceFile` and `CreateObject#ObjectInfoDisplayedOnWorkflowMetrics`?
If both tests are not related then we can forward the pr.

**Comment by diogomelo09** [comment]
ci:forward

**Comment by diogomelo09** [comment]
ci:forward:force

**Pull request closed without merging**